src/paho/mqtt/client.py

```diff
@@ -625,6 +624,196 @@ def __init__(self, client_id="", clean_session=None, userdata=None,
+                stacklevel=2,
+            )
+            self._sock.settimeout(value)
+
+        if value < 0:
+            raise ValueError("Keepalive must be >=0.")
+
+        self._keepalive = value
+
+    @property
+    def transport(self) -> str:
+        'Transport used for the connection, could be "tcp" or  "websockets".'
+        return self._transport
+
+    @transport.setter
+    def transport(self, value: str) -> None:
+        """
+        Update transport which should be "tcp" or "websockets".
+        This will only be used on future (re)connection. You should probably
+        use reconnect() to update the connection if established.
+        """
+        if value.lower() not in ("websockets", "tcp"):
+            raise ValueError(
+                f'transport must be "websockets" or "tcp", not {value}')
+
+        self._transport = value.lower()
+
+    @property
+    def protocol(self) -> int:
+        "Protocol version used (MQTT v3, MQTT v3.11, MQTTv5)"
+        return self.protocol
+
+    @property
+    def connect_timeout(self) -> float:
+        "Timeout used to establish the TCP (& TLS / websocket if enabled) in seconds"
+        return self._connect_timeout
+
+    @connect_timeout.setter
+    def connect_timeout(self, value: float):
+        "Change connect_timeout for future (re)connection"
+        if value <= 0.0:
+            raise ValueError("timeout must be a positive number")
+
+        self._connect_timeout = value
+
+    @property
+    def username(self) -> str | None:
+        "Return the username use to connect to MQTT broken or None if not credenials are used."
+        if self._username is None:
+            return None
+        return self._username.decode("utf-8")
+
+    @username.setter
+    def username(self, value: str | None) -> None:
+        """
+        Update username. This will only be used on future (re)connection. You should probably
+        use reconnect() to update the connection if established.
+        """
+        if value is None:
+            self._username = None
+        else:
+            self._username = value.encode("utf-8")
+
+    @property
+    def password(self) -> str | None:
+        "Return the password use to connect to MQTT broken or None if not password are used."
+        if self._password is None:
+            return None
+        return self._password.decode("utf-8")
+
+    @password.setter
+    def password(self, value: str | None) -> None:
+        """
+        Update password. This will only be used on future (re)connection. You should probably
+        use reconnect() to update the connection if established.
+        """
+        if value is None:
+            self._password = None
+        else:
+            self._password = value.encode("utf-8")
+
+    @property
+    def max_inflight_messages(self) -> int:
+        "Maximum number of message with QoS > 0 that can be part way through their network flow at once"
```

Comment:
```suggestion
        """Maximum number of messages with QoS > 0 that can be partway through the network flow at once"""
```
... but what does "partway through the network flow" mean? It's not quite clear.